eclipse/src/saros/Saros.java

```diff
@@ -260,29 +268,34 @@ private void stopLifeCycle() {
   }
 
   private void setupLoggers() {
-    /*
-     * HACK this is not the way OSGi works but it currently fulfill its
-     * purpose
-     */
-    final ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
 
     final boolean isDebugMode = Boolean.getBoolean("saros.debug") || isDebugging(); // $NON-NLS-1$
 
-    final String log4jPropertyFile =
-        isDebugMode
-            ? "saros_debug.log4j.properties" //$NON-NLS-1$
-            : "saros_release.log4j.properties"; //$NON-NLS-1$
-
     try {
-      // change the context class loader so Log4J will find the appenders
-      Thread.currentThread().setContextClassLoader(Saros.class.getClassLoader());
+      File mainLogDir = getStateLocation().toFile();
+      final String logDir = mainLogDir + File.separator + "log";
+
+      final Level logLevel = isDebugMode ? Level.ALL : Level.WARN;
 
-      PropertyConfigurator.configure(Saros.class.getClassLoader().getResource(log4jPropertyFile));
+      URL configFileU = getBundle().getResource(LOG4J2_CONFIG_FILENAME);
+      if (configFileU == null) {
+        throw new RuntimeException(
+            "Failed to get the log4j2 config file from the bundle class loader");
+      }
+      String configFileUri = configFileU.toString();
+
+      // set config file to be used
+      System.setProperty("log4j.configurationFile", configFileUri);
```

Comment:
While you can do this in your own application, we are a plugin. Isn't there a proper way of initializing log4j2 ?

Reply:
Sure, there should be a better solution.

We could:
- Force the reconfiguration via [`setConfigLocation`](https://logging.apache.org/log4j/2.x/faq.html#reconfig_from_code) (not really better)
- Migrate the whole [XML config to code](https://logging.apache.org/log4j/2.x/manual/customconfig.html#Reconfigure_Log4j_Using_ConfigurationBuilder_with_the_Configurator) that reconfigures log4j2 (better, but sharing the log configuration between STF, Eclipse, IntelliJ is more complex (via Core?))
- Don't reconfigure and use a [ConfigurationFactory](Initialize_Log4j_Using_ConfigurationBuilder_with_a_Custom_ConfigurationFactory). I think this is the best solution, but we have to verify whether the plugin's lifecycle and the log4j2's lifecycle are compatible and how to inject the log-dir into the factory.

Reply:
What I wanted to say is not to use the sys(tem) scope but a more specifc one, eg. ctx

See https://logging.apache.org/log4j/2.x/manual/configuration.html#PropertySubstitution

Reply:
Seems DataMap is the desired option. We could also use global system properties but I would suggest to name them saros.log4j.* to avoid name clashes.